backend/app/models.py

```diff
@@ -21,14 +21,31 @@ class Photo(models.Model):
     to the Map Square that the photo belongs to and the Photographer who
     took the photo.
     """
-
     # Use number and map_square to uniquely identify a photo
     # DO NOT use the database pk, as that may change as we rebuild the database
-    number = models.IntegerField(null=True)  # NOT the database PK, but rather based on shelfmark
+
+    # The source image filenames have the following structure:
+    # BHVP_PH_CetaitParis_DP_MAP-SQUARE-NUMBER_FOLDER-NUMBER_FILE-NUMBER
+
+    # The FILE-NUMBER comes in pairs, but numbered straight through. e.g., 
+    # BHVP_PH_CetaitParis_DP_0031_01_001 -- this is a scan of the slide, including its frame
+    # BHVP_PH_CetaitParis_DP_0031_01_002 -- this is a high quality scan of the photo itself
+    # We collapse these two into a single number: 1
+    # The next pair (01_003 and 01_004) become photo 2, etc.
+    number = models.IntegerField()
+
+    # Furthermore, many map squares contained multiple folders.
+    # These folders seemed meaningful -- they often represent boundary points between photographers.
+    folder = models.IntegerField()
+
+    # The original full filename of the SLIDE as provided to us by 
+    # e.g., "BHVP_PH_CetaitParis_DP_0031_01_001"
+    shelfmark = models.CharField(max_length=252)
```

Comment:
`integrate_new_images` doesn't currently save the shelfmark -- I totally forgot to do this. Let's not let this block the PR, I'll add this info in a follow-up PR.